config-linux.md

```diff
@@ -625,6 +625,22 @@ The following parameters can be specified to set up seccomp:
     * `SECCOMP_FILTER_FLAG_LOG`
     * `SECCOMP_FILTER_FLAG_SPEC_ALLOW`
 
+* **`listenerPath`** *(string, OPTIONAL)* - specifies the path of UNIX domain socket over which the runtime will send the [container process state](#containerprocessstate) data structure when the `SCMP_ACT_NOTIFY` action is used.
+    This socket MUST use `AF_UNIX` domain and `SOCK_STREAM` type.
+    The runtime MUST send exactly one [container process state](#containerprocessstate) per connection.
+    The connection MUST NOT be reused and it MUST be closed after sending a seccomp state.
+    If sending to this socket fails, the runtime MUST [generate an error](runtime.md#errors).
+    If the `SCMP_ACT_NOTIFY` action is not used this value is ignored.
+
+    The runtime sends the following file descriptors using `SCM_RIGHTS` and set their names in the `fds` array of the [container process state](#containerprocessstate):
+
+    * **`seccompFd`** (int, REQUIRED) is the seccomp file descriptor returned by the seccomp syscall.
```

Comment:
should *seccompFd* be optional as well?

Reply:
I don't mind if seccompFd is optional in the spec, but how would it work if the runtime does not pass the seccompFd? Do you want to make it optional for runtimes that don't want to implement seccomp-notify, or are you thinking of an alternative way to obtain the seccompFd?

For runtimes that use seccomp-notify but implements the seccomp agent internally, they can just not set the `listenerPath` field.

The "Container Process State" section is separate and does not mandate the seccompFd, but this section is in the description of "listenerPath".

Reply:
could it be possible to use it for retrieving `pidFd` (or anything more that is added in future) but not `seccompFd`?  e.g. what happens if the profile doesn't set any rule with `SECCOMP_ACTION_NOTIFY`?  Does the OCI runtime still need to create a  `seccompFd`?

Reply:
If the profile doesn't set any notify rule, we propose this field should be ignored by the runtime. See here: https://github.com/opencontainers/runtime-spec/pull/1074/files/d87dd02ab1435993552f72fd1d98c7779f67228a#diff-048d23d864e15683f516d2c1768965d546e87f8a59b2606cf2f2d52500ba5a32R633.

Our idea is to use this UNIX socket to send to a seccomp agent. Do you want to make this field more "generic"? Be aware that this is in the seccomp section, we can reuse the container state process in other parts of the spec, but this should be about seccomp.

Reply:
so do you think we should move the `listenerPath` outside the `seccomp` section?  I think it should be possible to use `pidFd` without seccomp notify, or seccomp in general

Reply:
> so do you think we should move the listenerPath outside the seccomp section?

I think we should not; it would make it more difficult to use in Kubernetes (see https://github.com/opencontainers/runtime-spec/pull/1073#issuecomment-720524314).

> I think it should be possible to use pidFd without seccomp notify, or seccomp in general

If you want a mechanism to get the pidFd without seccomp, I'd rather doing that separately in a separate field than `seccomp.listenerPath`, even if it reuses this new "Container Process State" type.

There can be only one seccomp agent receiving the seccompFd (by design from https://github.com/opencontainers/runtime-spec/pull/1073#discussion_r514382166). But I see use cases where several processes could receive the pidFd, so maybe it could be an array of listeners. Could this be a follow-up PR for this? In your use cases, would you want to receive the pidFd at container-create time or at container-start time?

Reply:
@giuseppe friendly ping? :)